src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud/config.jelly

```diff
@@ -27,6 +27,10 @@
 
     <f:validateButton title="${%Test Connection}" progress="${%Testing...}" method="testConnection" with="name,serverUrl,credentialsId,serverCertificate,skipTlsVerify,namespace" />
 
+    <f:entry title="${%Direct Connection}" field="directConnection">
+      <f:checkbox default="true"/>
```

Comment:
You are optimistic!

Reply:
Shall I switch to `false`? Assuming the feature works, from my perspective this is much more straight forward than configuring and using the HTTP(S) URL.

Reply:
The main drawback I see with this is that it won't work if the Jenkins instance is set to use a random port for inbound TCP. In case the master restarts, the agent won't be able to reconnect.

Reply:
I removed the default.

For our (Jenkinsfile Runner in Kubernetes) scenario Jenkins restarts are not relevant, but I'll try to test the behaviour.

How important is this topic for you? I mean the agents created by the Kubernetes plugin are typically not living very long.
I can also add some notes to the help.

Reply:
> the agents created by the Kubernetes plugin are typically not living very long

But they can survive a restart of the Jenkins master, which might occur within a matter of minutes after the agent pod has started. It is expected to reconnect to the restarted master, even with a random TCP port (`Jenkins.slaveAgentPort == 0`).

The [existing tests passed](https://kube.pse.beescloud.com/opensource/job/kubernetes-plugin/job/PR-602/1/testReport/org.csanchez.jenkins.plugins.kubernetes.pipeline/RestartPipelineTest/), but I suspect this was only because you made this field enabled by default in the GUI form whereas `KubernetesTestUtil.setupCloud` uses `KubernetesCloud.<init>(String)` which leaves it off.